### PR TITLE
added security to the project

### PR DESCRIPTION
**Pull request by kaiiiseeel**

### CODE REVIEW

**Review by Eidenn005 — APPROVED** [review]
LGTM

File: src/main/java/mate/academy/bookstore/config/SecurityConfig.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+import org.springframework.security.config.annotation.method.configuration.EnableMethodSecurity;
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.config.annotation.web.configurers.AbstractHttpConfigurer;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.security.web.SecurityFilterChain;
+
+@Configuration
+@EnableMethodSecurity
```
Comment (andrii-hoienko) [inline src/main/java/mate/academy/bookstore/config/SecurityConfig.java:17]:
```suggestion
@EnableMethodSecurity
@RequiredArgsConstructor
```

File: src/main/java/mate/academy/bookstore/config/SecurityConfig.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+public class SecurityConfig {
+    private final UserDetailsService userDetailsService;
+
+    public SecurityConfig(UserDetailsService userDetailsService) {
+        this.userDetailsService = userDetailsService;
+    }
```
Comment (andrii-hoienko) [inline src/main/java/mate/academy/bookstore/config/SecurityConfig.java]:
```suggestion

```

File: src/main/resources/db/changelog/changes/03-create-roles-table.yaml [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+                  type: bigint
+                  autoIncrement: true
+                  constraints:
+                    primaryKey: true
+              - column:
+                  name: name
+                  type: enum('ADMIN', 'USER')
+                  constraints:
+                    nullable: false
+                    unique: true
```
Comment (andrii-hoienko) [inline src/main/resources/db/changelog/changes/03-create-roles-table.yaml]:
```suggestion
                    unique: true
                    
```

File: src/main/resources/db/changelog/changes/04-create-users-roles-table.yaml [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+                    nullable: false
+                    foreignKeyName: fk_user_id
+                    references: users(id)
+              - column:
+                  name: role_id
+                  type: bigint
+                  constraints:
+                    nullable: false
+                    foreignKeyName: fk_role_id
+                    references: roles(id)
```
Comment (andrii-hoienko) [inline src/main/resources/db/changelog/changes/04-create-users-roles-table.yaml]:
```suggestion
                    references: roles(id)
                    
```

File: src/main/resources/db/changelog/changes/05-insert-roles-to-roles-table.yaml [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+            columns:
+              - column:
+                  name: name
+                  value: USER
+        - insert:
+            tableName: roles
+            columns:
+              - column:
+                  name: name
+                  value: ADMIN
```
Comment (andrii-hoienko) [inline src/main/resources/db/changelog/changes/05-insert-roles-to-roles-table.yaml]:
```suggestion
                  value: ADMIN
                  
```

File: src/main/resources/db/changelog/changes/06-insert-users-to-users-table.yaml [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+                  value: '$2a$10$su2EKXcxBLEYd4kHVM7J7eBoZytgas9LlpEMHtrLjygH/eemLnHye'
+              - column:
+                  name: first_name
+                  value: 'Bob'
+              - column:
+                  name: last_name
+                  value: 'Jackson'
+              - column:
+                  name: shipping_address
+                  value: '789 Oak St'
```
Comment (andrii-hoienko) [inline src/main/resources/db/changelog/changes/06-insert-users-to-users-table.yaml]:
```suggestion
                  value: '789 Oak St'
                  
```

**Review by andrii-hoienko — CHANGES_REQUESTED** [review]
Nice, left some minor comments

File: src/main/resources/db/changelog/changes/04-create-users-roles-table.yaml [hunk 7]
```diff
@@ -0,0 +1,22 @@
+databaseChangeLog:
+  - changeSet:
+      id: create-users-roles-table
+      author: Mykola Skrypalov
+      changes:
+        - createTable:
+            tableName: users_roles
```
Comment (vovasalyha) [inline src/main/resources/db/changelog/changes/04-create-users-roles-table.yaml:7]: Make user_id + role_id a composite PK to disallow duplicate entries.

**Review by vovasalyha — APPROVED** [review]
Approving, but take a look on one comment.

**Pull request merged**